src/hdusd/mx_nodes/nodes/base_node.py

```diff
@@ -26,6 +26,9 @@ class MxNodeInputSocket(bpy.types.NodeSocket):
     bl_label = "MX Input Socket"
 
     def draw(self, context, layout, node, text):
+        if not hasattr(node, 'nodedef'):    # handle MaterialX 1.37 nodes
```

Comment:
Move `hasattr(node, 'nodedef')` to special function in utils/mx.py and use this function in all required places.
Possible function name `mx_node_ok()`

Reply:
Or better create in mx_nodes/nodes/base_node.py function with name `is_node_ok()`